pandas/core/index.py

```diff
@@ -4184,12 +4184,12 @@ def copy(self, names=None, dtype=None, levels=None, labels=None,
             levels = self.levels
             labels = self.labels
             names = self.names
-        return MultiIndex(levels=levels,
-                          labels=labels,
-                          names=names,
-                          sortorder=self.sortorder,
-                          verify_integrity=False,
-                          _set_identity=_set_identity)
+        return self.__class__(levels=levels,
```

Comment:
this needs to be made consistent, e.g. use ``@constructor``, so this is a no-go (you may need to rejigger where the ``@constructor`` is used now as its ``MultiIndex.from_tuples``)